packages/console/src/components/CodeEditor/index.tsx

```diff
@@ -49,49 +53,52 @@ const CodeEditor = ({
-          spellCheck="false"
-          value={value}
-          onChange={handleChange}
-          onKeyDown={handleKeydown}
-        />
-        {/* SyntaxHighlighter will generate a <pre> tag and a inner <code> tag. Both have
-        inline-styles by default. Therefore, We can only use inline styles to customize them.
-        Some styles have to be applied multiple times to each of them for the sake of consistency. */}
-        <SyntaxHighlighter
-          wrapLongLines
-          codeTagProps={{
-            style: {
-              fontFamily: "'Roboto Mono', monospace", // Override default font-family of <code>
-            },
-          }}
-          customStyle={{
-            background: 'transparent',
-            fontSize: '14px',
-            margin: '0',
-            padding: '0',
-            borderRadius: '0',
-            wordBreak: 'break-all',
-            fontFamily: "'Roboto Mono', monospace", // Override default font-family of <pre>
-          }}
-          language={language}
-          style={theme}
-        >
-          {value}
-        </SyntaxHighlighter>
+    <>
+      <div className={classNames(styles.container, className)}>
+        <CopyToClipboard value={value} variant="icon" className={styles.copy} />
+        <div className={styles.editor}>
+          {/* SyntaxHighlighter is a readonly component, so a transparent <textarea> layer is needed
+      in order to support user interactions, such as code editing, copy-pasting, etc. */}
+          <textarea
+            ref={textareaRef}
+            autoCapitalize="off"
+            autoComplete="off"
+            autoCorrect="off"
+            data-gramm="false"
+            readOnly={isReadonly}
+            spellCheck="false"
+            value={value}
+            onChange={handleChange}
+            onKeyDown={handleKeydown}
+          />
+          {/* SyntaxHighlighter will generate a <pre> tag and a inner <code> tag. Both have
+      inline-styles by default. Therefore, We can only use inline styles to customize them.
+      Some styles have to be applied multiple times to each of them for the sake of consistency. */}
+          <SyntaxHighlighter
+            wrapLongLines
+            codeTagProps={{
+              style: {
+                fontFamily: "'Roboto Mono', monospace", // Override default font-family of <code>
+              },
+            }}
+            customStyle={{
+              background: 'transparent',
+              fontSize: '14px',
+              margin: '0',
+              padding: '0',
+              borderRadius: '0',
+              wordBreak: 'break-all',
+              fontFamily: "'Roboto Mono', monospace", // Override default font-family of <pre>
+            }}
+            language={language}
+            style={theme}
+          >
+            {value}
+          </SyntaxHighlighter>
+        </div>
       </div>
-    </div>
+      {hasError && errorMessage && <div className={styles.errorMessage}>{errorMessage}</div>}
```

Comment:
nit: the error message can be more friendly? like the exact position of error
(let's talk to designer and do it later)

Reply:
OK